### PR TITLE
Fix/compact tool wheel bottom reverse

### PR DESCRIPTION
**Pull request by CN-Zephyr**
## Summary by CodeRabbit

* **改进**
  * 紧凑输入工具轮盘在桌面端的 `viewport-fit` 视口适配更精准；桌面布局切换后轮盘状态同步更稳定。
  * 统一滚轮旋转、拖拽步进与充能方向的视觉方向计算，提升不同布局下的一致手感。
* **Bug 修复**
  * `viewport-fit` 下 `hidden/hidden-forward/hidden-backward` 槽位的过渡、变换与交互行为更正确；必要时增加遮罩回退，减少溢出显示异常。
* **测试**
  * 扩展桌面 `viewport-fit` 布局与交互用例，并对相关 `styles.css` 规则进行断言。

### CODE REVIEW

**Comment by CN-Zephyr** [comment]
Handled the CSS snapshot path issue in commit a562da16. I kept the underlying fix but avoided the direct import.meta.url/fileURLToPath form because this Vitest setup did not expose a usable file URL for the CSS raw path here; the helper now locates styles.css from either the package root or repo root before reading it. Verified with the targeted reversed-arc test and react-neko-chat typecheck.

File: frontend/react-neko-chat/src/App.tsx [hunk 1]
```diff
@@ -4093,10 +4120,22 @@ function CompactChatApp({
...
+    if (hasDesktopWorkArea) {
+      const workAreaBottom = workAreaY + workAreaHeight;
+      const wheelCenterScreenY = windowY + centerY;
+      const bottomGap = workAreaBottom - wheelCenterScreenY;
+      const bottomFlipThreshold = Math.max(
+        COMPACT_INPUT_TOOL_WHEEL_HOVER_RADIUS,
+        orbitRadius + (buttonSize / 2),
+      );
+      if (bottomGap < bottomFlipThreshold) return 'viewport-fit';
+    }
```
[the remark this answers is not in the document]
Comment (CN-Zephyr) [reply frontend/react-neko-chat/src/App.tsx:4139]: Partially fixed in commit e116fa38. The bottom-triggered reverse path now validates the viewport-fit arc against the horizontal work-area bounds before choosing viewport-fit, which covers the side-edge clipping case this comment points out. I intentionally did not require full vertical fit in this pre-expanded bottom case: when the compact capsule is still sitting at the taskbar, the renderer's current local fan rect can fail vertical bounds until the desktop shell expands/reserves upward, and enforcing the full check there regresses the intended bottom reverse behavior. Added a regression test for the side-edge case and re-ran App.test.tsx plus typecheck.

**Review by wehos — APPROVED** [review]
好的

File: frontend/react-neko-chat/src/App.test.tsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { readFileSync } from 'node:fs';
+import { resolve } from 'node:path';
```
[the remark this answers is not in the document]
Comment (CN-Zephyr) [reply frontend/react-neko-chat/src/App.test.tsx]: Handled in commit 291cd0f3. I kept the existing filesystem-based CSS assertion helper, but made the package contract explicit by adding @types/node as a devDependency and including node in tsconfig.app.json types. This fixes clean Linux/CI builds where the package no longer happens to inherit Node typings transitively. Verified npm run build plus the targeted viewport-fit wheel tests.
